tests/test_stats_grammar.py

```diff
@@ -244,3 +244,13 @@ def test_filtered_parse(self):
         prs = Parser(trace, filters={"cdev_state": 3})
         dfr_res = prs.solve("devfreq_out_power:freq")
         self.assertEquals(len(dfr_res), 1)
+
+    def test_no_events(self):
+        """Test trying to parse absent data"""
+        trace = trappy.FTrace()
+        prs = Parser(trace)
+
+        # cpu_frequency is an event we know how to parse, but it isn't present
+        # in the test trace.
+        self.assertRaisesRegexp(ValueError, "No events found for cpu_frequency",
+                                lambda: prs.solve("cpu_frequency:frequency"))
```

Comment:
don't use `lambda`, pass the args the arguments to `assertRaisesRegexp`:

``` python
self.assertRaisesRegexp(ValueError, "No events found for cpu_frequency",
                        prs.solve, "cpu_frequency:frequency")
```

or use the context manager:

``` python
with self.assertRaisesRegexp(ValueError, "No events found for cpu_frequency"):
    prs.solve("cpu_frequency:frequency")
```